packages/@lwc/synthetic-shadow/src/3rdparty/inner-text.ts

```diff
@@ -0,0 +1,243 @@
+        if (!nodeIsBeingRendered(computedStyle)) {
+            if (tagName === 'SELECT' || tagName === 'DATALIST') {
+                // the select is either: .visibility != 'visible' or .display === hidden, therefore this select should
+                // not display any value.
+                return [];
+            }
+
+            return result;
+        }
+
+        // 5. If node is a br element, then append a string containing a single U+000A LINE FEED (LF) character to items.
+        if (tagName === 'BR') {
+            result.push('\n');
+        }
+
+        // 6. If node's computed value of 'display' is 'table-cell', and node's CSS box is not the last 'table-cell' box of its enclosing 'table-row' box, then append a string containing a single U+0009 CHARACTER TABULATION (tab) character to items.
+        if (computedStyle.display === 'table-cell') {
+            // omitting case: and node's CSS box is not the last 'table-cell' box of its enclosing 'table-row' box
+            result.push('\t');
+        }
+
+        // 7. If node's computed value of 'display' is 'table-row', and node's CSS box is not the last 'table-row' box of the nearest ancestor 'table' box, then append a string containing a single U+000A LINE FEED (LF) character to items.
+        if (computedStyle.display === 'table-row') {
+            // omitting case: and node's CSS box is not the last 'table-row' box of the nearest ancestor 'table' box
+            result.push('\n');
+        }
+
+        // 8. If node is a p element, then append 2 (a required line break count) at the beginning and end of items.
+        if (tagName === 'P') {
+            result.unshift(2);
+            result.push(2);
+        }
+
+        // 9. If node's used value of 'display' is block-level or 'table-caption', then append 1 (a required line break count) at the beginning and end of items.
+        if (computedStyle.display === 'block' || computedStyle.display === 'table-caption') {
+            result.unshift(1);
+            result.push(1);
+        }
+    } else if (node.nodeType === TEXT_NODE) {
+        result.push(getTextNodeInnerText(node));
+    }
+
+    return result;
+}
+
+/**
+ * innerText spec: https://html.spec.whatwg.org/multipage/dom.html#the-innertext-idl-attribute
+ */
+export function getInnerText(this: Element): string {
+    const thisComputedStyle = getElementComputedStyle(this);
+    // 1. If this is not being rendered or if the user agent is a non-CSS user agent, then return this's descendant text content.
+    if (!nodeIsBeingRendered(thisComputedStyle)) {
+        return this.textContent || ''; // textContentGetterPatched.call(this);
+    }
+
+    const selectionState = getSelectionState(this);
+
+    // 2. Let results be a new empty list.
+    let results: InnerTextCollectionResult[] = [];
+    // 3. For each child node node of this:
+    const childNodes = this.childNodes;
```

Comment:
Here is where the heart of this implementation is. Add a note here that `this.childNodes` will excludes shadow trees because `Node.childNodes` is patched.